OpenVote/src/com/openvote/LoginServlet.java

```diff
@@ -29,7 +34,12 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
 	
     public void doPost(HttpServletRequest req, HttpServletResponse resp)
             throws IOException {
-    	
+
+		UserService userService=UserServiceFactory.getUserService();
+		User user=userService.getCurrentUser();
+		
+		if (user!= null && isAdminLoggedIn()) {
```

Comment:
Isn't it the same to just call userService.isUserAdmin() instead of having this isAdminLoggedIn() method? If the user is null, the if condition will short-circuit and not NullPointer error on that